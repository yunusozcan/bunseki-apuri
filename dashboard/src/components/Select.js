import React, {useEffect, useState} from 'react';
import "react-dates/initialize";
import {DateRangePicker} from "react-dates";
import "react-dates/lib/css/_datepicker.css";
import '../stylesheets/components/Select.css';

function Select(props) {
    const {parentCallback, defaultData} = props;
    const [startDate, setStartDate] = useState(null);
    const [endDate, setEndDate] = useState(null);
    const [focusedInput, setFocusedInput] = useState(null);
    const handleDatesChange = ({startDate, endDate}) => {
        setStartDate(startDate);
        setEndDate(endDate);
        if (startDate && endDate) {
            parentCallback({
                startDate: startDate.format(),
                endDate: endDate.format()
            });
        }
    };

    useEffect(() => {
        if (defaultData) {
            setStartDate(null);
            setEndDate(null);
        }
    }, [defaultData]);

    return (
        <div className="App-select">
            <DateRangePicker
                startDate={startDate}
                startDateId={"tata-start-date"}
                endDate={endDate}
                endDateId={"tata-end-date"}
                onDatesChange={handleDatesChange}
                focusedInput={focusedInput}
                onFocusChange={focusedInput => setFocusedInput(focusedInput)}
                orientation={window.innerWidth <= 768 ? "vertical" : "horizontal"}
                openDirection={"down"}
                noBorder={true}
                small={window.innerWidth <= 414 ? true : false}
                withFullScreenPortal={window.innerWidth <= 768 ? true : false}
                isOutsideRange={() => false}
            />
        </div>
    );
}

export default Select;
