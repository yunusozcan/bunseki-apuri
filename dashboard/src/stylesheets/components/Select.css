.App-select {
    flex-grow: 10;
    display: flex;
    text-align: center;
    justify-content: center;
    align-items: center;
}

.DateInput {
    width: 100px;
}

.DateInput__small {
    width: 80px;
}

.DateRangePicker_picker {
    z-index: 99;
}

.DayPickerNavigation_button__verticalDefault {
    width: 30%;
}

@media only screen and (max-width: 320px) {
    .DateInput__small {
        width: 70px;
    }

    .DateInput_input__small {
        font-size: 13px;
    }

}
