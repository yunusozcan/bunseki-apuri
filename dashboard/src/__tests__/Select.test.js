import React from 'react';
import Select from '../components/Select';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
        .create(<Select defaultData={true} parentCallback={() => { console.log('Test');}}>Facebook</Select>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
