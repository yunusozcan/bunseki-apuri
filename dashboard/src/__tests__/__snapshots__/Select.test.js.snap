// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App-select"
>
  <div
    className="DateRangePicker DateRangePicker_1"
  >
    <div>
      <div
        className="DateRangePickerInput DateRangePickerInput_1"
      >
        <div
          className="DateInput DateInput_1"
        >
          <input
            aria-describedby="DateInput__screen-reader-message-tata-start-date"
            aria-label="Start Date"
            autoComplete="off"
            className="DateInput_input DateInput_input_1"
            disabled={false}
            id="tata-start-date"
            name="tata-start-date"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Start Date"
            readOnly={false}
            required={false}
            type="text"
            value=""
          />
          <p
            className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
            id="DateInput__screen-reader-message-tata-start-date"
          >
            Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
          </p>
        </div>
        <div
          aria-hidden="true"
          className="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
          role="presentation"
        >
          <svg
            className="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
            focusable="false"
            viewBox="0 0 1000 1000"
          >
            <path
              d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
            />
          </svg>
        </div>
        <div
          className="DateInput DateInput_1"
        >
          <input
            aria-describedby="DateInput__screen-reader-message-tata-end-date"
            aria-label="End Date"
            autoComplete="off"
            className="DateInput_input DateInput_input_1"
            disabled={false}
            id="tata-end-date"
            name="tata-end-date"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="End Date"
            readOnly={false}
            required={false}
            type="text"
            value=""
          />
          <p
            className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
            id="DateInput__screen-reader-message-tata-end-date"
          >
            Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
